src/providers/DocumentSymbolProvider.ts

```diff
@@ -74,14 +74,12 @@ export class DocumentSymbolProvider implements vscode.DocumentSymbolProvider {
 						let type_lines: string[] = [];
 						let found_end_type: boolean = false;
 						let exploded_str: string = '';
-						while(found_end_type == false) {
```

Comment:
This was the problem for the hung symbol provider not parsing fully. To fix it removed the while, as the condition and edge cases and permutations possible is just a lot. So we try up to 1024 lines when we encounter "type" and if there is  a type longer than 1k, well, best effort...